README.md

```diff
@@ -20,9 +18,217 @@ Or install it yourself as:
 
     $ gem install plaid
 
+The gem supports Ruby 2.x only.
+
 ## Usage
 
-TODO: Write usage instructions here
+First of all, the [official Plaid documentation](http://plaid.com/docs/) is your friend. 
+The gem wraps the API which is fully described there.
+
+The RubyDoc for the gem is available [here](http://plaid.github.io/plaid-ruby).
+
+### Configuring access to Plaid
+
+Configure the gem with your client id, secret, and the environment you would like to use.
+
+```ruby
+Plaid.config do |p|
+  p.client_id = '<<< Plaid provided client ID >>>'
+  p.secret = '<<< Plaid provided secret key >>>'
+  p.env = :tartan  # or :api for production
+end
+```
+
+### Creating a new User
+
+```ruby
+user = Plaid::User.create(:connect, 'wells', 'plaid_test', 'plaid_good')
+```
+
+This call will do a `POST /connect`. The response will contain account information and transactions 
+for last 30 days, which you can find in `user.accounts` and `user.initial_transactions`, accordingly.
+
+If the authentication requires a pin, you can pass it as a named parameter:
+
+```ruby
+user = Plaid::User.create(:income, 'usaa', 'plaid_test', 'plaid_good', 'wells', pin: '1234')
+```
+
+To add options such as `login_only` or `webhook`, use `options` argument:
+
+```ruby
+user = Plaid::User.create(:connect, 'wells', 'plaid_test', 'plaid_good', 
+                          options: { login_only: true, webhook: 'https://example.org/callbacks/plaid')
+```
+
+The first argument for `Plaid::User.create` is always a product you want to add the user to (like,
+`:connect`, `:auth`, `:info`, `:income`, or `:risk`). The user object is bound to the product, and subsequent
+calls like `user.update` or `user.delete` are done for this product (i.e., `PATCH /info` and `DELETE /info`
+for `:info`). If you need to change the current product, use `User.for_product` or even `User.upgrade`.
```

Comment:
can you document an example of calling `User.upgrade`?

Also, do we need both `for_product` and `upgrade`? Can we simplify and just have one way for users to do this?